### ISSUE TITLE
Consider depending on async: ^2.2.0

### ISSUE THREAD

**Issue by caiopo** [marcelgarus/list_diff#4]
By using the latest flutter_test version (bundled with flutter 1.22) and depending on list_diff, version solving fails due to conflicts in `async` version.

Seems to be possible to solve by increasing the `async` version range to `^2.2.0`. I believe it's unlikely that `async` will add a breaking change in a non-major version.

Version solving log:
```
Because every version of flutter_test from sdk depends on test_api 0.2.19-nullsafety.2 which depends on async >=2.5.0-nullsafety <2.5.0, every version of flutter_test from sdk requires async >=2.5.0-nullsafety <2.5.0.
And because list_diff >=1.1.1 depends on async >=2.2.0 <2.5.0, flutter_test from sdk is incompatible with list_diff >=1.1.1.
And because implicitly_animated_list 1.0.2 depends on list_diff ^1.1.2 and no versions of implicitly_animated_list match >1.0.2 <2.0.0, flutter_test from sdk is incompatible with implicitly_animated_list ^1.0.2.
So, because my_app depends on both implicitly_animated_list ^1.0.2 and flutter_test any from sdk, version solving failed.
pub get failed (1; So, because my_app depends on both implicitly_animated_list ^1.0.2 and flutter_test any from sdk, version solving failed.)
```

Would be happy to help with a PR.

PS: for those with the same problem as me, adding this to pubspec.yaml temporarily fixes the issue:

```
dependency_overrides:
  async: 2.5.0-nullsafety.1
```

**Comment by MarcelGarus** [comment]
I relaxed the dependency to `async: ^2.2.0` and published it as 1.3.2 on pub.

**Comment by caiopo** [comment]
Thanks! :)
